src/plugin/luapi_application.h

```diff
@@ -233,13 +234,75 @@ static int applib_changeCurrentPageBackground(lua_State* L) {
     return 1;
 }
 
+/**
+ * Change color of a specified tool or of the current tool
+ */
+static int applib_changeColor(lua_State* L) {
+
+    // discard any extra arguments passed in
+    lua_settop(L, 1);
+    luaL_checktype(L, 1, LUA_TTABLE);
+
+    lua_getfield(L, 1, "selection"); /* either true or false, for changing selection color
+                                       defaults to true*/
+    lua_getfield(L, 1, "tool");      /* "pen", "hilighter", "text"
+                                      "select_rect", "select_object", "select_region"
+                                      if omitted, current Tool is used */
+    lua_getfield(L, 1, "color");     // an RGB hex code defining the color
+    // stack now has following:
+    //    1 = {["color"] = 0xff00ff, ["tool"] = "PEN", ["selection"] = false}
+    //   -3 = false
+    //   -2 = "pen"
+    //   -1 = 0xff0077
+
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* ctrl = plugin->getControl();
+    ToolHandler* toolHandler = ctrl->getToolHandler();
+
+    bool selection = false;
+    if (lua_isboolean(L, -3)) {
+        selection = lua_toboolean(L, -3);
+    }
+
+    ToolType toolType = toolHandler->getToolType();
+    const char* toolStr = luaL_optstring(L, -2, nullptr);
+    if (toolStr != nullptr) {
+        toolType = toolTypeFromString(StringUtils::toLowerCase(toolStr));
+    }
+
+    if (toolType == TOOL_NONE) {
+        g_warning("tool \"%s\" is not valid or no tool has been selected", toolTypeToString(toolType).c_str());
+        lua_pop(L, 3);
+        return 0;
+    }
+
+    int color = 0x000000;
+    if (lua_isinteger(L, -1)) {
+        color = lua_tointeger(L, -1);
```

Comment:
Check if the color is valid RGBA.

Reply:
RGB you mean?

Reply:
Yeah, RGB.

It's too easy to mix up the formats, so we should replace all (un)signed int colors with typedefs like `rgb_t` or `rgba_t` in the future.